test/test_nn.py

```diff
@@ -5397,6 +5397,26 @@ def test_smoothl1loss_negative_beta_not_supported(self):
         with self.assertRaises(RuntimeError):
             F.smooth_l1_loss(torch.randn(2, 2), torch.randn(2, 2), beta=-1.0)
 
+    def test_smoothl1loss_backward_zero_beta_cpu(self):
+        input = torch.randn(300, 256, requires_grad=True, device='cpu')
+        target = input.detach()
+
+        loss = F.smooth_l1_loss(input, target, beta=0.0, reduction='sum')
+        loss.backward()
+
+        grad_max_abs = input.grad.abs().max().item()
+        self.assertLessEqual(grad_max_abs, 1.0)
+
+    def test_smoothl1loss_backward_zero_beta_cuda(self):
```

Comment:
Should I also add @onlyCPU for the cpu test or not?

Reply:
Not that is not necessary.

Reply:
Ho this decorate is for the per-device tests.
Can you move your test to the per-Device class? It is the ones just above that take `(self, device)` as argument instead of just self. 
From there, you can have one test that use the device passed in instead of having one test for cpu and one test for cuda.